src/Util/ClassUtils.php

```diff
@@ -110,4 +117,31 @@ public static function generateProxyClassName($className, $proxyNamespace)
     {
         return rtrim($proxyNamespace, '\\') . '\\' . Proxy::MARKER . '\\' . ltrim($className, '\\');
     }
+
+    /**
+     * Check if the type is an enum type or type containing an enum type
+     *
+     * @param ReflectionNamedType|ReflectionUnionType|ReflectionIntersectionType|null $type
+     *
+     */
+    public static function containsEnumType($type): bool
```

Comment:
I would type the argument as `ReturnType|null`, to match the one of `ReflectionMethod::getReturnType`

Reply:
Union types are only allowed since PHP 8.0.

The PHP min version for this project is 7.1.